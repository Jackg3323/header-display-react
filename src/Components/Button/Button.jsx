import PropTypes from "prop-types";

export default function Button({ txt, clickHandler: ClickHandler}) {
  return (
    <button
      type="button"
      onClick={ClickHandler}
      className = ` rounded text-white px-4 py-2`
    >
      {txt}
    </button>
  );
}

Button.propTypes = {
  txt: PropTypes.string.isRequired,
  bgColor: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  bgColor: "green",
};
