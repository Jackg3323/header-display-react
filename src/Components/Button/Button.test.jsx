import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import Button from "./Button";

it("renders correctly", () => {
  const tree = renderer
    .create(<Button clickHandler={() => {}} txt="Click Me" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("calls the change handler whenever the button is clicked", () => {
  const mockClickHandler = jest.fn();

  render(<Button changeHandler={mockClickHandler} />);
  userEvent.onClick(screen.getByRole("button"));

  expect(mockClickHandler).toHaveBeenCalled();
});
