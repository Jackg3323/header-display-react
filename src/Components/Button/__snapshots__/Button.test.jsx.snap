// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="rounded text-white bg-green-500 px-4 py-2"
  onClick={[Function]}
  type="button"
>
  Click Me
</button>
`;
